udata/tests/test_discussions.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from datetime import datetime
+
+from flask import url_for, Blueprint
+
+from udata.api import api
+from udata.models import db
+
+from udata.core.metrics.models import WithMetrics
+from udata.core.user.views import blueprint as user_bp
+
+from udata.core.discussions.models import Discussion, Message
+from udata.core.discussions.metrics import DiscussionsMetric
+from udata.core.discussions.api import DiscussionsAPI
+
+from .api import APITestCase
+from .factories import faker, UserFactory
+
+
+bp = Blueprint('test_discussions', __name__)
+ns = api.namespace('test', 'A test namespace')
+
+
+class Fake(WithMetrics, db.Document):
+    name = db.StringField()
+    owner = db.ReferenceField('User')
+    organization = db.ReferenceField('Organization')
+
+
+class FakeDiscussion(Discussion):
+    subject = db.ReferenceField(Fake)
```

Comment:
@noirbizarre there is now an issue with the subject in tests because the newly introduced `ModelChoiceField` accepts only defined models in `forms.DiscussionCreateForm` and `Fake` is not one of these models… Do you have any idea how to handle it?

Note that otherwise it works fine locally.